kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/BatchTableWriter.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo;
+import com.google.cloud.bigquery.LoadJobConfiguration;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.gson.Gson;
+import jdk.nashorn.internal.runtime.regexp.joni.exception.ValueException;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeoutException;
+
+public class BatchTableWriter implements Runnable {
+    private Gson gson;
+
+    private BlobInfo blobInfo;
+    private Storage storage;
+    private String sourceUri;
+
+    private BigQuery bigQuery;
+    private TableId tableId;
+    private LoadJobConfiguration loadJobConfiguration;
+
+    public BatchTableWriter(String bucketName, String blobName, Storage storage,
+                            TableId tableId, BigQuery bigQuery) {
+        gson = new Gson();
+
+        BlobId blobId = BlobId.of(bucketName, blobName+".json");
+        blobInfo = BlobInfo.newBuilder(blobId).setContentType("text/json").build();
+        this.storage = storage;
+        this.sourceUri = "gs://" + bucketName + "/" + blobName + ".json";
+
+        this.tableId = tableId;
+        this.bigQuery = bigQuery;
+
+        // Check if the table specified exists
+        try {
+            if (!bigQuery.getTable(tableId).exists()) {
+                throw new ValueException("");
+            }
+        } catch (ValueException | NullPointerException exception) {
+            throw new ValueException(exception +
+                    String.format("Table id for table {} does not exist"), tableId.getTable());
+        }
+
+        this.loadJobConfiguration =
+                LoadJobConfiguration.builder(tableId, sourceUri)
+                .setFormatOptions(FormatOptions.json())
+                .setCreateDisposition(JobInfo.CreateDisposition.CREATE_NEVER)
+                .build();
+    }
+
+    @Override
+    public void run() {
+        //todo implement
+    }
+
+    void transferBlobToBigQuery() throws InterruptedException, TimeoutException {
+        Job loadJob = bigQuery.create(JobInfo.of(loadJobConfiguration));
+        try {
+            loadJob = loadJob.waitFor();
+        } catch (InterruptedException ie) {
+            throw new InterruptedException(ie + "Transfer from GCS blob to BigQuery unsuccessful.");
```

Comment:
`throw new InterruptedException("Transfer from GCS blob to BigQuery unsuccessful.", ie);`

Reply:
Also, recommend including the bucket, blob, and table id (or full loaded configuration) in exception.

Reply:
It appears that there is no constructor for the InterruptedException class that takes a string and a Throwable. Let me know how you would like me to handle this.